wave_vision/include/wave/vision/matcher/descriptor_matcher.hpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file
+ * Base class from which descriptor matchers can be derived.
+ * @ingroup vision
+ */
+#ifndef WAVE_VISION_DESCRIPTOR_MATCHER_HPP
+#define WAVE_VISION_DESCRIPTOR_MATCHER_HPP
+
+/** C++ Headers */
+#include <exception>
+
+/** Third Party Headers */
+#include <opencv2/opencv.hpp>
+
+/** Libwave Headers */
+#include "wave/utils/utils.hpp"
+
+namespace wave {
+/** @addtogroup vision
+ *  @{ */
+
+/** Representation of a generic descriptor matcher.
+ *
+ *  Internally, this class, and all derived classes are wrapping various
+ *  descriptor matchers implemented in OpenCV. Further reference on descriptor
+ *  matchers can be found [here][opencv_descriptor_matchers].
+ *
+ *  [opencv_descriptor_matchers]:
+ *  http://docs.opencv.org/trunk/db/d39/classcv_1_1DescriptorMatcher.html
+ */
+
+class DescriptorMatcher {
+ protected:
+    /** Destructor */
+    ~DescriptorMatcher() = default;
```

Comment:
This class is not trivial anyway; in this case the subtle differences between `{}` and `= default` aren't really important except `default` makes your intentions clear.

The notable thing here is Nav made the destructor `protected` and _non_-`virtual`. This means even though this class is meant to be inherited, derived classes will not be destroyed in an unsafe way (see https://accu.org/index.php/journals/1896).